keel-sql/src/main/resources/db/changelog/20201116-foreign-keys.yml

```diff
@@ -0,0 +1,290 @@
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-environment_resource
+    author: fletch
+    changes:
+    - addForeignKeyConstraint:
+        baseTableName: environment_resource
+        baseColumnNames: environment_uid
+        constraintName: fk_environment_resource_environment
+        referencedTableName: environment
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - addForeignKeyConstraint:
+        baseTableName: environment_resource
+        baseColumnNames: resource_uid
+        constraintName: fk_environment_resource_resource
+        referencedTableName: resource
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: foreign-keys-on-resource_last_checked
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from resource_last_checked where not exists (select 1 from resource where resource.uid = resource_last_checked.resource_uid)
+    - addForeignKeyConstraint:
+        baseTableName: resource_last_checked
+        baseColumnNames: resource_uid
+        constraintName: fk_resource_last_checked_resource
+        referencedTableName: resource
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+- changeSet:
+    id: resource_uid-on-unhappy_veto
+    author: fletch
+    changes:
+    - addColumn:
+        tableName: unhappy_veto
+        columns:
+        - column:
+            name: resource_uid
+            type: char(26)
+    - modifySql:
+        dbms: mysql
+        append:
+          value: " first"
+- changeSet:
+    id: foreign-keys-on-unhappy_veto
+    author: fletch
+    changes:
+    - sql:
+        sql: delete from unhappy_veto where not exists (select 1 from resource where resource.id = unhappy_veto.resource_id)
+    - sql:
+        sql: update unhappy_veto set resource_uid = (select uid from resource where id = unhappy_veto.resource_id)
+    - addNotNullConstraint:
+        tableName: unhappy_veto
+        columnName: resource_uid
+        columnDataType: char(26)
+    - addForeignKeyConstraint:
+        baseTableName: unhappy_veto
+        baseColumnNames: resource_uid
+        constraintName: fk_unhappy_veto_resource
+        referencedTableName: resource
+        referencedColumnNames: uid
+        referencesUniqueColumn: true
+        onDelete: CASCADE
+    - dropColumn:
+        tableName: unhappy_veto
+        columnName: resource_id
+    - dropColumn:
+        tableName: unhappy_veto
+        columnName: application
+- changeSet:
+    id: resource_uid-on-unhealthy
+    author: fletch
+    changes:
+    - addColumn:
+        tableName: unhealthy
+        columns:
+        - column:
+            name: resource_uid
+            type: char(26)
+    - modifySql:
+        dbms: mysql
+        append:
+          value: " first"
```

Comment:
Note: this is how you get a new column in the first spot in a table in mySQL. It does not support Liquibase's `position` or `beforeColumn` properties, only `afterColumn`.